hs_composite_resource/tests/test_composite_resource.py

```diff
@@ -2676,3 +2680,97 @@ def test_unzip_aggregation_overwrite(self):
+
+    def test_zip_aggregation_virtual_folder_3(self):
+        """Test that we can zip a netcdf aggregation that exists in a fileset folder.
+           The aggregation zip file becomes a resource file and part of the fileset aggregation.
+        """
+        self.create_composite_resource()
+        self.assertEqual(NetCDFLogicalFile.objects.count(), 0)
+        self.assertEqual(FileSetLogicalFile.objects.count(), 0)
+        folder = "fs-folder"
+        ResourceFile.create_folder(resource=self.composite_resource, folder=folder)
+        # create a netcdf aggregation inside a folder- which we will test for zipping
+        self.add_file_to_resource(file_to_add=self.netcdf_file, upload_folder=folder)
+        self.assertEqual(self.composite_resource.files.count(), 1)
+        nc_res_file = self.composite_resource.files.first()
+        NetCDFLogicalFile.set_file_type(resource=self.composite_resource, file_id=nc_res_file.id, user=self.user)
+        self.assertEqual(NetCDFLogicalFile.objects.count(), 1)
+        self.assertEqual(self.composite_resource.files.count(), 2)
+        # set the folder to fileset aggregation
+        FileSetLogicalFile.set_file_type(resource=self.composite_resource, folder_path=folder, user=self.user)
+        self.assertEqual(FileSetLogicalFile.objects.count(), 1)
+        fs_aggr = FileSetLogicalFile.objects.first()
+        self.assertEqual(fs_aggr.files.count(), 0)
+        nc_aggr = NetCDFLogicalFile.objects.first()
+        zip_file_name = 'nc_aggr'
+        zip_aggregation_virtual_folder(user=self.user, res_id=self.composite_resource.short_id,
+                                       aggregation_name=nc_aggr.aggregation_name, output_zip_fname=zip_file_name)
+        self.assertEqual(self.composite_resource.files.count(), 3)
+        zip_res_file = ResourceFile.get(resource=self.composite_resource, file=f"{zip_file_name}.zip", folder=folder)
+        self.assertTrue(zip_res_file.exists)
+        self.assertEqual(fs_aggr.files.count(), 1)
+        self.assertEqual(zip_res_file.logical_file_type_name, "FileSetLogicalFile")
+        self._test_zip_file_contents(zipfile=zip_res_file, aggregation=nc_aggr)
+
+    def _test_zip_file_contents(self, zipfile, aggregation):
+        temp_zip_file = get_file_from_irods(resource=self.composite_resource, file_path=zipfile.storage_path)
+        aggr_files = []
+        for f in aggregation.files.all():
+            aggr_files.append(f.file_name)
+        aggr_files.append(os.path.basename(aggregation.map_file_path))
+        aggr_files.append(os.path.basename(aggregation.metadata_file_path))
+        with ZipFile(temp_zip_file) as zipfile:
+            files = zipfile.namelist()
+            self.assertEqual(len(files), len(aggr_files))
+            self.assertEqual(len(files), 4)
+            for f in files:
+                self.assertIn(f, aggr_files)
+        if os.path.exists(temp_zip_file):
+            shutil.rmtree(os.path.dirname(temp_zip_file))
```

Comment:
I think line 2775 is not needed and line 2776 can go within the block of line 2769.